src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/FinalLocalVariableCheck.java

```diff
@@ -667,6 +697,9 @@ private static boolean isLoopAst(int ast) {
         /** Contains definitions of uninitialized variables. */
         private final Deque<DetailAST> uninitializedVariables = new ArrayDeque<>();
 
+        /** Whether there is a {@code break} or {@code return} in the scope. */
+        private boolean containsBreakOrReturn;
```

Comment:
> You are correct. continue should be taken into consideration as well.

I say let's split this into new issue. This check is already complex enough and let's just get this PR done.

> One thing we need to confirm: should we concern about unreachable statement?

Unreachable statements won't change the fact that a variable can be `final` or not, right?
Our main concern is can the variable be declared `final` if it is compilable.
If user has other problems or makes changes to the code after that, it is not our concern.

> The only case I could find it different now is break with a label:

I forgot about this type of break. There will also be `continue` with a label too.
Unless @romani specifies otherwise, lets ignore this for now.

> I assume the only case we should raise a violation is smth like this ... And the check won't raise that violation

It is definately weird code.
I'll this up to @romani if we should fix or make a new issue or ignore.

> Is it really necessary to handle such weird cases like that? 

From our side, it is probably too much work to handle every single case imaginable.
From users' side, it would make this check all more amazing if it could find everything. :)

> To sum up, IMO, we don't need to handle break and return in a different way. Please provide a counterexample if I am wrong.

Thanks for all to think about. I will do my own testing and get back to you. I have just one idea I want to test out.
I didn't see anything stand out in regression report.